cosmo_tester/test_suites/test_manager_upgrade/manager_upgrade_base.py

```diff
@@ -257,17 +260,57 @@ def _load_private_ip_from_env(self, workdir):
         env = self._bootstrap_local_env(workdir)
         return env.outputs()['private_ip']
 
+    def _load_public_ip_from_env(self, workdir):
+        env = self._bootstrap_local_env(workdir)
+        return env.outputs()['manager_ip']
+
     def bootstrap_manager(self):
-        self.bootstrap_blueprint = self.get_bootstrap_blueprint()
-        inputs_path = self.manager_cfy._get_inputs_in_temp_file(
-            self.manager_inputs, self._testMethodName)
 
-        self.manager_cfy.bootstrap(self.bootstrap_blueprint,
-                                   inputs_file=inputs_path)
+        with self._bootstrap_old_version():
+            self.upgrade_manager_ip = self._load_public_ip_from_env(
+                    os.getcwd())
+            self.manager_private_ip = self._load_private_ip_from_env(
+                    os.getcwd())
+        self.cfy.init(reset_config=True).wait()
+        self.cfy.use(self.upgrade_manager_ip).wait()
 
-        self.upgrade_manager_ip = self.manager_cfy.get_management_ip()
-        self.manager_private_ip = self._load_private_ip_from_env(
-            self.cfy_workdir)
+    @contextmanager
+    def _bootstrap_old_version(self):
```

Comment:
i dont really like this use of a contextmanager - i think there's too much logic happening here for it to be a reasonable use. there's a whole bootstrap going on here :D
I'd have one method which bootstraps old version (up to "try" part), another which cleans up (the finally part), and just call them with a try-finally clause from the outside.
if there were more places calling this and requiring the same logic maybe it'd make sense doing it this way but thats not the case here